src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java

```diff
@@ -67,11 +70,15 @@ public void processQuestionnaireLinked(
         messageTimestamp);
   }
 
-  private void checkQidNotLinkedToAnotherCase(UacDTO uac, UacQidLink uacQidLink) {
+  private void logErrorIfQidIsLinkedToAnotherCase(
+      UacDTO uac, UacQidLink uacQidLink, String newLinkedCaseId) {
     if (uacQidLink.getCaze() != null
         && !uacQidLink.getCaze().getCaseId().equals(UUID.fromString(uac.getCaseId()))) {
-      throw new RuntimeException(
-          "UacQidLink already linked to case id: " + uacQidLink.getCaze().getCaseId());
+      log.with("uac_qid_link_id", uacQidLink.getId())
```

Comment:
I'd probably prefer it if we logged the QID - I don't think it's a security risk to have it in the logs. The QID tells us quite a lot without having to access the database

Reply:
Fair point, we do get some important information from the QID itself. 👍